File: 905. Sort Array By Parity/solution.cpp
// two pointer
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) {
        int evenIndex = 0;
        int oddIndex = nums.size()-1;
        
        while(evenIndex<oddIndex){
            if(nums[evenIndex]%2 > nums[oddIndex]%2){
                int tem = nums[evenIndex];
                nums[evenIndex] = nums[oddIndex];
                nums[oddIndex] = tem;
            }
            if(nums[evenIndex]%2 == 0) evenIndex++;
            if(nums[oddIndex]%2 == 1) oddIndex--;
        }
        
        return nums;
    }

};

File: 905. Sort Array By Parity/solution.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        evenIndex = 0
        oddIndex = len(nums) -1
        
        while(evenIndex<oddIndex):
            if nums[evenIndex]%2 > nums[oddIndex]%2:
                tem =  nums[evenIndex]
                nums[evenIndex] = nums[oddIndex]
                nums[oddIndex] = tem
            
            if nums[evenIndex]%2 == 0:
                evenIndex +=1
            
            if nums[oddIndex]%2 == 1:
                oddIndex -=1
                
        return nums
